src/main/java/com/github/tomakehurst/wiremock/admin/model/SnapshotSpec.java

```diff
@@ -0,0 +1,72 @@
+package com.github.tomakehurst.wiremock.admin.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.tomakehurst.wiremock.extension.Parameters;
+import com.github.tomakehurst.wiremock.matching.MultiValuePattern;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Encapsulates options for generating and outputting StubMappings
+ */
+public class SnapshotSpec {
+    // Whitelist requests to generate StubMappings for
+    private ServeEventRequestFilters filters;
+    // Headers from the request to include in the stub mapping, if they match the corresponding matcher
+    private RequestPatternTransformer captureHeaders;
+    // Criteria for extracting body from responses
+    private ResponseDefinitionBodyMatcher extractBodyCriteria;
+    // How to format StubMappings in the response body
+    // Either "full" (meaning return an array of rendered StubMappings) or "ids", which returns an array of UUIDs
+    private SnapshotOutputFormat outputFormat;
+    // Whether to persist stub mappings
+    private boolean persist = true;
+    // Whether duplicate requests should be recorded as scenarios or just discarded
+    private boolean repeatsAsScenarios = false;
+    // Stub mapping transformers
+    private List<String> transformers;
+    // Parameters for stub mapping transformers
+    private Parameters transformerParameters;
+
+    @JsonCreator
+    public SnapshotSpec(
+        @JsonProperty("filters") ServeEventRequestFilters filters ,
+        @JsonProperty("captureHeaders") Map<String, MultiValuePattern> captureHeaders,
+        @JsonProperty("extractBodyCriteria") ResponseDefinitionBodyMatcher extractBodyCriteria,
+        @JsonProperty("outputFormat") SnapshotOutputFormat outputFormat,
+        @JsonProperty("persist") JsonNode persistNode,
```

Comment:
You're right, it'd be better if this wasn't coupled to Jackson. Fixed in 331ba0a148ffd8e75f061c788c21ee692b36dd14